api/test/e2enew/global_rule/global_rule_test.go

```diff
@@ -0,0 +1,201 @@
+			base.RunTestCase(tc)
+		},
+		Entry("create route", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"name": "route1",
+				"uri": "/hello",
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "` + base.UpstreamIp + `",
+						"port": 1981,
+						"weight": 1
+					}]
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
+		Entry("create global rule", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/global_rules/1",
+			Body: `{
+				"plugins": {
+					"response-rewrite": {
+						"headers": {
+							"X-VERSION":"1.0"
+						}
+					},
+					"uri-blocker": {
+						"block_rules": ["select.+(from|limit)", "(?:(union(.*?)select))"]
+					}
+				}
+			}`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"X-VERSION\":\"1.0\"",
+		}),
+		Entry("verify route with header", base.HttpTestCase{
+			Object:        base.APISIXExpect(),
+			Method:        http.MethodGet,
+			Path:          "/hello",
+			ExpectStatus:  http.StatusOK,
+			ExpectBody:    "hello world",
+			ExpectHeaders: map[string]string{"X-VERSION": "1.0"},
+		}),
+		Entry("verify route that should be blocked", base.HttpTestCase{
+			Object:        base.APISIXExpect(),
+			Method:        http.MethodGet,
+			Path:          "/hello",
+			Query:         "name=%3Bselect%20from%20sys",
+			ExpectStatus:  http.StatusForbidden,
+			ExpectHeaders: map[string]string{"X-VERSION": "1.0"},
+		}),
+		Entry("update route with same plugin response-rewrite", base.HttpTestCase{
+			Object: base.ManagerApiExpect(),
+			Method: http.MethodPut,
+			Path:   "/apisix/admin/routes/r1",
+			Body: `{
+				"name": "route1",
+				"uri": "/hello",
+				"plugins": {
+					"response-rewrite": {
+						"headers": {
+							"X-VERSION":"2.0"
+						}
+					}
+				},
+				"upstream": {
+					"type": "roundrobin",
+					"nodes": [{
+						"host": "` + base.UpstreamIp + `",
+						"port": 1981,
+						"weight": 1
+					}]
+				}
+			 }`,
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "\"X-VERSION\":\"2.0\"",
+		}),
+		Entry("verify route that header should override", base.HttpTestCase{
+			Object:        base.APISIXExpect(),
+			Method:        http.MethodGet,
+			Path:          "/hello",
+			ExpectStatus:  http.StatusOK,
+			ExpectBody:    "hello world",
+			ExpectHeaders: map[string]string{"X-VERSION": "2.0"},
+		}),
+		Entry("delete global rule", base.HttpTestCase{
+			Object:       base.ManagerApiExpect(),
+			Method:       http.MethodDelete,
+			Path:         "/apisix/admin/global_rules/1",
+			Headers:      map[string]string{"Authorization": base.GetToken()},
+			ExpectStatus: http.StatusOK,
+		}),
```

Comment:
we could add a test case to check the delete

Reply:
I did this test in the CURD section.
[https://github.com/apache/apisix-dashboard/blob/665c3480a22e8bd1d4a232aa813ad214156af5b9/api/test/e2enew/global_rule/global_rule_test.go#L83-L89](https://github.com/apache/apisix-dashboard/blob/665c3480a22e8bd1d4a232aa813ad214156af5b9/api/test/e2enew/global_rule/global_rule_test.go#L83-L89)

Reply:
Oh,it's not to test the deletion itself, but to make sure that it is really deleted in DP.

Reply:
Hi, @nic-chen, sorry for the delayed response.
This will be executed in the `AfterSuite` phase, i.e. he will remove the content involved after the test is completed for each type of data.
[https://github.com/apache/apisix-dashboard/blob/665c3480a22e8bd1d4a232aa813ad214156af5b9/api/test/e2enew/global_rule/global_rule_suite_test.go#L34-L38](https://github.com/apache/apisix-dashboard/blob/665c3480a22e8bd1d4a232aa813ad214156af5b9/api/test/e2enew/global_rule/global_rule_suite_test.go#L34-L38)